ios/sdk/src/callkit/JMCallKitProxy.swift

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright @ 2018-present Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CallKit
+import Foundation
+
+/// JitsiMeet CallKit proxy
+@available(iOS 10.0, *)
+@objc public final class JMCallKitProxy: NSObject {
+    
+    override private init() {}
+    
+    // MARK: - CallKit proxy
+    
+    internal static let cxProvider: CXProvider = {
+        let config = CXProviderConfiguration(localizedName: "")
+        let provider = CXProvider(configuration: config)
+        return provider
+    }()
+    
+    internal static let cxCallController: CXCallController = {
+        return CXCallController()
+    }()
+    
+    internal static let callKitNotifier: JMCallKitNotifier = {
+        return JMCallKitNotifier()
+    }()
+    
+    internal static var cxProviderConfiguration: CXProviderConfiguration? {
+        didSet {
+            guard let providerConfiguration = cxProviderConfiguration else { return }
+            cxProvider.configuration = providerConfiguration
+            cxProvider.setDelegate(callKitNotifier, queue: nil)
+        }
+    }
+    
+    /// Enables the proxy in between callkit and the consumers of the SDK
+    /// Default to enabled, set to false when you don't want to use callkit
+    @objc public static var enabled: Bool = true {
+        didSet {
+            if enabled == false {
+                cxProvider.setDelegate(nil, queue: nil)
+            }
+        }
+    }
+    
+    @objc public static func hasProviderBeenConfigurated() -> Bool {
```

Comment:
The word "configurated" bothers me, so I'll change it to "configured".

In JavaScript and Java we'd have gone with `isProviderConfigured`. It's shorter (i.e. easier to read), so I'll adopt that name.